server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5367,12 +5414,17 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
         }
 
         Long serviceOfferingId = cmd.getServiceOfferingId();
+        Long overrideDiskOfferingId = cmd.getOverrideDiskOfferingId();
 
         ServiceOffering serviceOffering = _entityMgr.findById(ServiceOffering.class, serviceOfferingId);
         if (serviceOffering == null) {
             throw new InvalidParameterValueException("Unable to find service offering: " + serviceOfferingId);
         }
 
+        if (serviceOffering.getDiskOfferingStrictness() && overrideDiskOfferingId != null) {
+            throw new InvalidParameterValueException(String.format("Cannot user override disk offering id %d since provided service offering is strictly mapped to its disk offering", overrideDiskOfferingId));
```

Comment:
```suggestion
            throw new InvalidParameterValueException(String.format("Cannot override disk offering id %d since provided service offering is strictly mapped to its disk offering", overrideDiskOfferingId));
```